### PR TITLE
redo of flo's tag branch and merge

### PR DESCRIPTION
**Pull request by JannikMichel**
Pull Request for Issue #5474 .

### Summary of Changes
I try to finish the great work from @jrseliga and @florian1995.
@florian1995 fixed the problem that when you deselect for example all tags the list won't refresh. 

The following test instructions are also from @jrseliga.

### Testing Instructions
<h3>Administrator</h3>

<h3>Articles</h3>

  1. Create 3 articles:
    - Article 1 - Category 1 - Tag 1 - Author 1 - Public
    - Article 2 - Category 2 - Tag 1 & Tag 2 - Author 2 - Registered
    - Article 3 - Category 2 - Tag 2 - Author 1 - Super User
  2. View Articles
    - Use Search Tools

Just with a simple structure like this, there is a large number of potential test cases. Please refer to the New behavior section for an overview of the filtering logic

<h3>Featured Articles</h3>

  1. View Featured Articles
  Use Search Tools

<h3>Categories</h3>

  1. View Categories
  -Use Search Tools
Developed@icampus

### CODE REVIEW

**Comment by Schmidie64** [comment]
I have tested this item :white_check_mark: successfully on cedb5eef1cc77666f541a740ef7b3da6939d6a3d<br /><br />@icampus
Everything works how it should be. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17668">issues.joomla.org/tracker/joomla-cms/17668</a>.</sub>

**Comment by eXsiLe95** [comment]
I have tested this item :white_check_mark: successfully on cedb5eef1cc77666f541a740ef7b3da6939d6a3d<br /><br />
## Tested:

### Testing 

1. Created multiple articles (like described)
2. Went to article overview
3. Used article filters

### Before patch

Only one option per filter can be selected. Multiple filters are useable, but not multiple options on one filter.

### With patch

Multiple options on a single filter can be selected. Works as described and expected.

### Suggestions

You can not see what field matches a filter. Therefore, the initial value could be displayed instead of dummy text "Type or select some options". Maybe, you should use CSS placeholder instead of HTML value, too.

Tested @iCampus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17668">issues.joomla.org/tracker/joomla-cms/17668</a>.</sub>

File: administrator/components/com_content/models/articles.php [hunk 1]
```diff
@@ -99,26 +99,38 @@ protected function populateState($ordering = 'a.id', $direction = 'desc')
 		$search = $this->getUserStateFromRequest($this->context . '.filter.search', 'filter_search');
 		$this->setState('filter.search', $search);
```
Comment (Quy) [inline administrator/components/com_content/models/articles.php:101]: Remove blank line.

File: administrator/components/com_content/models/articles.php [hunk 2]
```diff
@@ -99,26 +99,38 @@ protected function populateState($ordering = 'a.id', $direction = 'desc')
...
 
-		$access = $this->getUserStateFromRequest($this->context . '.filter.access', 'filter_access');
-		$this->setState('filter.access', $access);
-
-		$authorId = $app->getUserStateFromRequest($this->context . '.filter.author_id', 'filter_author_id');
-		$this->setState('filter.author_id', $authorId);
```
Comment (Quy) [inline administrator/components/com_content/models/articles.php]: Remove blank line.

File: administrator/components/com_content/models/articles.php [hunk 3]
```diff
@@ -99,26 +99,38 @@ protected function populateState($ordering = 'a.id', $direction = 'desc')
...
 
 		$published = $this->getUserStateFromRequest($this->context . '.filter.published', 'filter_published', '');
 		$this->setState('filter.published', $published);
```
Comment (Quy) [inline administrator/components/com_content/models/articles.php]: Remove blank line.

File: administrator/components/com_content/models/articles.php [hunk 4]
```diff
@@ -21,7 +21,7 @@ class ContentModelArticles extends JModelList
 	/**
 	 * Constructor.
 	 *
-	 * @param   array  $config  An optional associative array of configuration settings.
+	 * @param   array $config An optional associative array of configuration settings.
```
Comment (Quy) [inline administrator/components/com_content/models/articles.php]: Per [Joomla's Coding Manual](https://developer.joomla.org/coding-standards/docblocks.html):
`The minimum whitespace between any text elements, such as tags, variable types, variable names and tag descriptions, is two real spaces.`

File: administrator/components/com_content/models/forms/filter_articles.xml [hunk 5]
```diff
@@ -22,34 +22,37 @@
 		<field
 			name="category_id"
 			type="category"
+			multiple="true"
+			class="multipleCategories"
```
Comment (Quy) [inline administrator/components/com_content/models/forms/filter_articles.xml]: `The four attributes name, type, label and description should be written in this order and at the top of the element definition.`

File: administrator/components/com_content/models/forms/filter_articles.xml [hunk 6]
```diff
@@ -22,34 +22,37 @@
...
 			label="JOPTION_FILTER_CATEGORY"
 			description="JOPTION_FILTER_CATEGORY_DESC"
 			extension="com_content"
 			onchange="this.form.submit();"
 			published="0,1,2"
 			>
```
Comment (Quy) [inline administrator/components/com_content/models/forms/filter_articles.xml]: Please read the coding style for XML when element is empty: https://developer.joomla.org/coding-standards/xml.html

File: administrator/components/com_content/models/forms/filter_articles.xml [hunk 7]
```diff
@@ -22,34 +22,37 @@
...
 			type="author"
+			multiple="true"
+			class="multipleAuthors"
 			label="COM_CONTENT_FILTER_AUTHOR"
 			description="COM_CONTENT_FILTER_AUTHOR_DESC"
 			onchange="this.form.submit();"
-			>
-			<option value="">JOPTION_SELECT_AUTHOR</option>	
-		</field>
+					>
```
Comment (Quy) [inline administrator/components/com_content/models/forms/filter_articles.xml]: Fix tab and coding style.

File: administrator/components/com_content/models/forms/filter_articles.xml [hunk 8]
```diff
@@ -65,12 +68,14 @@
...
 			name="tag"
 			type="tag"
+			multiple="true"
+			class="multipleTags"
 			label="JOPTION_FILTER_TAG"
 			description="JOPTION_FILTER_TAG_DESC"
 			mode="nested"
 			onchange="this.form.submit();"
 			>
-			<option value="">JOPTION_SELECT_TAG</option>
```
Comment (Quy) [inline administrator/components/com_content/models/forms/filter_articles.xml]: Fix coding style.

File: administrator/components/com_content/models/forms/filter_featured.xml [hunk 9]
```diff
@@ -1,140 +1,146 @@
...
-			>
-			<option value="">JOPTION_SELECT_PUBLISHED</option>
-		</field>
+        <field
+                name="published"
+                type="status"
+                label="COM_CONTENT_FILTER_PUBLISHED"
+                description="COM_CONTENT_FILTER_PUBLISHED_DESC"
+                onchange="this.form.submit();"
+        >
```
Comment (Quy) [inline administrator/components/com_content/models/forms/filter_featured.xml]: Fix tab.

File: administrator/components/com_content/models/forms/filter_featured.xml [hunk 10]
```diff
@@ -1,140 +1,146 @@
...
-			onchange="this.form.submit();"
-			>
-			<option value="">JOPTION_SELECT_CATEGORY</option>
-		</field>
+        <field
+                name="category_id"
+                type="category"
+                multiple="true"
+                class="multipleCategories"
+                label="JOPTION_FILTER_CATEGORY"
```
Comment (Quy) [inline administrator/components/com_content/models/forms/filter_featured.xml]: Fix order.

File: administrator/components/com_content/models/forms/filter_articles.xml [hunk 11]
```diff
@@ -15,7 +15,7 @@
 			label="COM_CONTENT_FILTER_PUBLISHED"
 			description="COM_CONTENT_FILTER_PUBLISHED_DESC"
 			onchange="this.form.submit();"
-			>
+            >
```
Comment (Quy) [inline administrator/components/com_content/models/forms/filter_articles.xml]: Use tabs and not spaces. Same issue in the other places.

**Comment by eXsiLe95** [comment]
I have tested this item :white_check_mark: successfully on 52d7b39c1d67f0bee6fae9f223a92ca9eae935f7<br /><br />## Tested:

### Testing 

1. Created multiple articles (like described)
2. Went to article overview
3. Used article filters

### Before patch

Only one option per filter can be selected. Multiple filters are useable, but not multiple options on one filter.

### With patch

Multiple options on a single filter can be selected. Works as described and expected.

### Notes

Since the update, every field has its own placeholder. Good job!

Tested @iCampus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17668">issues.joomla.org/tracker/joomla-cms/17668</a>.</sub>

**Comment by DrDreave** [comment]
I have tested this item :white_check_mark: successfully on 52d7b39c1d67f0bee6fae9f223a92ca9eae935f7<br /><br />
# Operating System
* Joomla! 3.8.0-beta3-dev
* PHP 5.6.2
* MySQLi 5.5.38
* Apache/2.2.29 (Unix)

# Steps
* Open article administration
* Use search tools

# Test before patch
* Single filter selection
Single filter selection

# Test after patch
* Multi filter selection for category, access, author tag
Multi filter selection

Tested @icampus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17668">issues.joomla.org/tracker/joomla-cms/17668</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by infograf768** [comment]
This has broken multilang frontend!

**Comment by infograf768** [comment]
Not only multilang. any menu item displaying a category does not display any article.
`There are no articles in this category. If subcategories display on this page, they may contain articles.`

**Comment by infograf768** [comment]
Issue:
https://github.com/joomla/joomla-cms/issues/17694

**Comment by infograf768** [comment]
Patch here:
https://github.com/joomla/joomla-cms/pull/17697 corrects this issue.

Please test.

**Comment by infograf768** [comment]
This contains a useless string
See https://github.com/joomla/joomla-cms/pull/17835

**Pull request merged**